src/v3.rs

```diff
@@ -237,20 +230,29 @@ impl Message {
 }
 
 impl Email {
-    /// Construct a new email type.
-    pub fn new() -> Email {
-        Email::default()
-    }
-
-    /// Set the address for this email.
-    pub fn set_email(mut self, email: &str) -> Email {
-        self.email = String::from(email);
-        self
+    /// Construct a new email type with name set as None.
+    ///
+    /// ```rust
+    /// use sendgrid::v3::Email;
+    ///
+    /// let my_email = Email::new("test@mail.com".to_owned());
+    /// ```
+    pub fn new<S: Into<String>>(email: S) -> Email {
+        Email {
+            email: email.into(),
+            name: None,
+        }
     }
 
     /// Set an optional name.
-    pub fn set_name(mut self, name: &str) -> Email {
-        self.name = Some(String::from(name));
+    ///
+    /// ```rust
+    /// use sendgrid::v3::Email;
+    ///
+    /// let my_email = Email::new("test@mail.com".to_owned()).set_name("My Name".to_owned());
```

Comment:
We can remove the `to_owned` if we have a parameter that must implement `Into<String>` right?